app/javascript/crayons/MentionAutocompleteTextArea/MentionAutocompleteTextArea.jsx

```diff
@@ -54,284 +73,311 @@ const replaceTextArea = ({
-    // Construct the new textArea content with selected username inserted
-    const textWithSelection = `${textContent.substring(
-      0,
-      selectionInsertIndex,
-    )}${username} ${textContent.substring(comboboxRef.current.selectionStart)}`;
-
-    // Clear the current search
-    setSearchTerm('');
-    setUsers([]);
-    setAriaHelperText('');
-
-    // Update the text area value
-    setTextContent(textWithSelection);
-
-    // Switch back to the plain text input
-    comboboxRef.current.classList.add('hidden');
-    plainTextAreaRef.current.classList.remove('hidden');
-
-    // Update the cursor to directly after the selection (+2 accounts for the @ sign, and adding a space after the username)
-    const newCursorPosition = selectionInsertIndex + username.length + 2;
-    setCursorPosition(newCursorPosition);
-  };
-
-  useLayoutEffect(() => {
-    const { current: comboboxTextArea } = comboboxRef;
-    const { current: plainTextArea } = plainTextAreaRef;
-
-    if (comboboxTextArea && plainTextArea) {
-      replaceTextArea({
-        originalNodeToReplace: replaceElement,
-        plainTextArea,
-        comboboxTextArea,
-      });
-    }
-  }, [replaceElement]);
-
-  return (
-    <Fragment>
-      <div aria-live="polite" class="screen-reader-only">
-        {ariaHelperText}
-      </div>
-
-      <Combobox
-        id="combobox-container"
-        onSelect={handleSelect}
-        className="crayons-autocomplete"
-      >
-        <ComboboxInput
-          ref={comboboxRef}
-          value={textContent}
-          data-mention-autocomplete-active="true"
-          as="textarea"
-          autocomplete={false}
-          onChange={handleTextInputChange}
-        />
-
-        <textarea
-          data-mention-autocomplete-active="true"
-          ref={plainTextAreaRef}
-          onChange={handleTextInputChange}
-          value={textContent}
-        />
-
-        {searchTerm && (
-          <ComboboxPopover
-            ref={popoverRef}
-            className="crayons-autocomplete__popover absolute"
-            id="mention-autocomplete-popover"
-            style={{
-              top: `calc(${dropdownPositionPoints.y}px + 1.5rem)`,
-              left: `${dropdownPositionPoints.x}px`,
+    }, [replaceElement]);
+
+    return (
+      <Fragment>
+        <div aria-live="polite" class="screen-reader-only">
+          {ariaHelperText}
+        </div>
+
+        <Combobox
+          ref={containerRef}
+          id="combobox-container"
+          onSelect={handleSelect}
+          className={`crayons-autocomplete${autoResize ? ' h-100' : ''}`}
+        >
+          <ComboboxInput
+            {...autocompleteInputProps}
+            aria-label="Mention user"
+            ref={comboboxRef}
+            value={textContent}
+            data-mention-autocomplete-active="true"
+            as="textarea"
+            autocomplete={false}
+            onChange={(e) => {
+              onChange?.(e);
```

Comment:
a question for my own edification/curiosity...how does this:
```javascript
onChange?.(e)
```
work?! (I haven't written JS in some time, so maybe this is just some fancy syntax I haven't seen that often 🙈)

Reply:
It's a newer language feature called [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining). It means only run the function if it exists.

Reply:
Nice! It _looked_ exactly like the [safe navigation operator in ruby](https://mitrev.net/ruby/2015/11/13/the-operator-in-ruby/) to me, and it seems like that's exactly what this is! Thanks for the quick explanation, @nickytonline + @aitchiss :)